py-polars/polars/expr/expr.py

```diff
@@ -3984,8 +3984,11 @@ def map_batches(
         A reasonable use case for `map` functions is transforming the values
         represented by an expression using a third-party library.
 
-        Read more in `the book
-        <https://pola-rs.github.io/polars/user-guide/expressions/user-defined-functions>`_.
+        .. warning::
+            If you are looking to map a function over a window function or group_by
+            context, refer to :func:`map_elements` instead.
+            Read more in `the book
+            <https://pola-rs.github.io/polars/user-guide/expressions/user-defined-functions>`_.
```

Comment:
I don't think this note is visibible enough, so I'm suggesting to move it to a warnings directive